structlog/structlog.go

```diff
@@ -0,0 +1,104 @@
+package structlog
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+var mapregex *regexp.Regexp
+var blacklistregexp []*regexp.Regexp
+
+func init() {
```

Comment:
Let's not (ever, if we can avoid it) use `init()` functions... they complicate the code's lifecycle, and tbh I have no idea what value they actually add. I'm [not alone](https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html) here either.

In this case, you can easily define them inline:

```
var mapregex *regexp.Regexp = regexp.MustCompile(`mapstructure:"([^"]+)"`)
var blacklistregexp []*regexp.Regexp = []*regexp.Regexp{
  regexp.MustCompile("password"),
}
```

(ironically, I understand that the article also recommends "no package-level variables" too. I think he's right about that in general, except that immutable/threadsafe ones like these are ok. If _all_ package-level vars were bad, then we'd never declare `const`s either... which I think is a little ridiculous)